### DISCUSSION TITLE
Byzantine chess

### DISCUSSION THREAD

**Discussion by hangarter** [fairy-stockfish/Fairy-Stockfish#309] [Variants]
Hi!

I'm trying to find an engine that I can use for Byzantine chess (https://www.chessvariants.com/play/jocly/byzantine-chess)
Reading the docs I couls see that potentially fairy-stockfish supports new variants, could you please help me to find if it's possible to use fairy-stockfish for this chess variant?

Cheers

**Comment by ianfab** [comment]
Hi. In principle Fairy-Stockfish does allow to add new variants, even at runtime. However, the topology of the board can not be changed, so only rectangular boards are supported. Therefore, variants like circular/byzantine chess or hex unfortunately are out of scope.

**Reply by WALEX2000** [reply]
Follow up question:
Does that mean that boards with holes are also not allowed?
And what about unmovable pieces?
And lastly, any way to represent Fog of War?

**Reply by PraseodymiumSpike** [reply]
Unmovable pieces are allowed. For example, see the Offering in Chak: https://www.pychess.org/variants/chak. The other things are not supported as far as I know.
